### ISSUE TITLE
compile error on macOS

### ISSUE THREAD

**Issue by Fabian188** [sergiud/SuiteSparse#6]
On macOS with Apple clang version 14.0.0 I get the error

```
Consolidate compiler generated dependencies of target umfpack_int
[ 31%] Building C object CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o
/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source/umf_cholmod.c:175:9: error: implicit declaration of function 'cholmod_analyze' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    L = CHOLMOD_analyze (S, &cm) ;
        ^
/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source/umf_cholmod.c:24:29: note: expanded from macro 'CHOLMOD_analyze'
#define CHOLMOD_analyze     cholmod_analyze
                            ^
/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source/umf_cholmod.c:175:9: note: did you mean 'cholmod_scale'?
/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source/umf_cholmod.c:24:29: note: expanded from macro 'CHOLMOD_analyze'
#define CHOLMOD_analyze     cholmod_analyze
                            ^
/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Include/cholmod_matrixops.h:122:5: note: 'cholmod_scale' declared here
int cholmod_scale
    ^
/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source/umf_cholmod.c:175:7: warning: incompatible integer to pointer conversion assigning to 'cholmod_factor *' (aka 'struct cholmod_factor_struct *') from 'int' [-Wint-conversion]
    L = CHOLMOD_analyze (S, &cm) ;
```
The `-Werror` puzzles me, as it seems it could be avoided, however it is unclear how to avoid it. The compile call is

```
/Library/Developer/CommandLineTools/usr/bin/cc -DBLAS_UNDERSCORE -DDINT -Dumfpack_EXPORTS -I/Users/fwein/Downloads/SuiteSparse-cmake/AMD/Include -I/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Include -I/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source -I/Users/fwein/Downloads/SuiteSparse-cmake/build/include/SuiteSparse/UMFPACK -I/Users/fwein/Downloads/SuiteSparse-cmake/build/CHOLMOD/Include -I/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Cholesky -I/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Core -I/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Include -I/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/MatrixOps -I/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Modify -I/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Supernodal -I/Users/fwein/Downloads/SuiteSparse-cmake/build/include/SuiteSparse/CHOLMOD -I/Users/fwein/Downloads/SuiteSparse-cmake/SuiteSparse_config -I/Users/fwein/Downloads/SuiteSparse-cmake/build/include/SuiteSparse -I/Users/fwein/Downloads/SuiteSparse-cmake/build/include/SuiteSparse/AMD -O3 -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.3 -fPIC -fvisibility=hidden -MD -MT CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o -MF CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o.d -o CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o -c /Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source/umf_cholmod.c
```

I don't know if this can be avoided by the project or if is an issue of the original sparse suite. The current sparse suite 6.0 beta compiles easily.

**Comment by sergiud** [comment]
Could you provide a full reproducer? While I don't use macOS, the project's [macos-12 Github workflow](https://github.com/sergiud/SuiteSparse/actions/workflows/macos.yml) currently running AppleClang 13.1.6 is perfectly happy. Thus, at this point I must assume there's something funky with your setup.

**Comment by Fabian188** [comment]
You are right, it is an option.

When I enable `WITH_CHECK`, it fails. What does this option mean anyway? The cmake option help is not really helpful ("Enable the Check module").

For me this hint helped and I'm fine - thanks :)

**Comment by Fabian188** [comment]
BTW, all on but only `WITH_CHOLESKY`
does also not work

```
[ 19%] Building CXX object CMakeFiles/spqr.dir/SPQR/Source/spqr_1colamd.cpp.o
/Users/fwein/Downloads/SuiteSparse-cmake/SPQR/Source/spqr_1colamd.cpp:564:13: error: use of undeclared identifier 'cholmod_l_amd'
            cholmod_l_amd (AT, NULL, 0, (Long *) (Q1fill + n1cols), cc) ;
            ^
/Users/fwein/Downloads/SuiteSparse-cmake/SPQR/Source/spqr_1colamd.cpp:581:18: error: use of undeclared identifier 'cholmod_l_analyze_p2'
            Sc = cholmod_l_analyze_p2 (FALSE, AT, NULL, NULL, 0, cc) ;
                 ^
/Users/fwein/Downloads/SuiteSparse-cmake/SPQR/Source/spqr_1colamd.cpp:606:13: error: use of undeclared identifier 'cholmod_l_colamd'; did you mean 'cholmod_l_ccolamd'?
            cholmod_l_colamd (AT, NULL, 0, TRUE,
            ^~~~~~~~~~~~~~~~
            cholmod_l_ccolamd
/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Include/cholmod_camd.h:61:5: note: 'cholmod_l_ccolamd' declared here
int cholmod_l_ccolamd (cholmod_sparse *, SuiteSparse_long *, size_t,
    ^
/Users/fwein/Downloads/SuiteSparse-cmake/SPQR/Source/spqr_1colamd.cpp:606:44: error: cannot initialize a parameter of type 'long *' with an rvalue of type 'int'
            cholmod_l_colamd (AT, NULL, 0, TRUE,
                                           ^~~~
/Users/fwein/Downloads/SuiteSparse-cmake/SPQR/Include/spqr.hpp:67:14: note: expanded from macro 'TRUE'
#define TRUE 1
             ^
/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Include/cholmod_camd.h:62:23: note: passing argument to parameter here
    SuiteSparse_long *, SuiteSparse_long *, cholmod_common *) ;
                      ^
4 errors generated.

```

But again, this is on issue, just an observation.

**Comment by sergiud** [comment]
The purpose of the Check module is explained in the official SuiteSparse guide:

https://github.com/sergiud/SuiteSparse/blob/5fa24ce1d67472e0e5cffd79d79b16000cd0f3c4/CHOLMOD/Doc/CHOLMOD_UserGuide.tex#L992-L998

I can reproduce the `WITH_CHOLESKY=OFF` build failure on my Linux machine. Let me look into the issue.

**Comment by sergiud** [comment]
> When I enable `WITH_CHECK`, it fails.

I'm not sure I can follow because the option is enabled by default. There is something else you have enabled. Please provide your exact CMake command-line.

**Comment by Fabian188** [comment]
I'm sorry - that was a stupid logical type: I meant, it fails when I **disable** WITH_CHECK. This is the only option I trigger.

**Comment by sergiud** [comment]
I see, thanks. I can now reproduce the initial problem as well.

**Comment by sergiud** [comment]
Both issues should be fixed in 5c1d50c26a05bccb58547bf550c4e13aef9fa1a9. Let me know if there any problems left.
